Microservices/Repositories/BaseCrudRepository.cs

```diff
@@ -0,0 +1,122 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microservices.Helpers;
+
+namespace Microservices.Repositories
+{
+    public abstract class BaseCrudRepository<TContext, TEntity> : IUnitOfWorkRepository<TEntity>
```

Comment:
In general for base CRUD operations it is not necessary to use repository pattern and db context can be used directly in services.

Reply:
Will discuss it offline.